src/main/java/bio/terra/service/dataset/flight/ingest/IngestPopulateFileStateFromFlightMapStep.java

```diff
@@ -66,6 +80,31 @@ public StepResult doSkippableStep(FlightContext context) {
     return StepResult.getStepResultSuccess();
   }
 
+  /**
+   * Filter the stream so that only files that need to be ingested are ingested, but also save the
+   * files that have already been ingested.
+   *
+   * @param bulkLoadModel A candidate file to maybe ingest
+   * @param collector A place to put the FileModels that already exist
+   * @return A Stream of BulkLoadFileModel if it needs to be ingested, or an empty Stream if not.
+   */
+  private Stream<BulkLoadFileModel> fileNotIngested(
+      BulkLoadFileModel bulkLoadModel, List<FileModel> collector) {
+    int depth = bulkLoadModel.getTargetPath().replaceAll("[^/]*", "").length();
+    boolean pathExists =
```

Comment:
This looks like it requires is to lookup the file twice (once to see if it exists and once to read it)

Could we combine this into a single call where we attempt to fetch and if it's not null (or empty) then add it.

That would likely be far more performant